File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/components/validators/RequiredValidator.java
package cz.cvut.fel.matyapav.afandroid.components.validators;

import android.view.View;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.RadioButton;
import android.widget.RadioGroup;

import cz.cvut.fel.matyapav.afandroid.components.parts.AFField;
import cz.cvut.fel.matyapav.afandroid.components.parts.ValidationRule;
import cz.cvut.fel.matyapav.afandroid.enums.SupportedWidgets;
import cz.cvut.fel.matyapav.afandroid.utils.Localization;
import cz.cvut.fel.matyapav.afandroid.utils.Utils;

/**
 * Created by Pavel on 14.02.2016.
 */
public class RequiredValidator implements AFValidator {

    @Override
    public boolean validate(AFField field, StringBuilder errorMsgs, ValidationRule rule) {
        boolean validationIsFine = true;
        if(Utils.isFieldWritable(field.getWidgetType().getWidgetName())){
            EditText textfield = (EditText) field.getFieldView();
            if (textfield.getText() == null || textfield.getText().toString().isEmpty()) {
                validationIsFine = false;
            }
        }
        if(field.getWidgetType() == SupportedWidgets.OPTION){
            RadioGroup radioGroup = (RadioGroup) field.getFieldView();
            if(radioGroup.getCheckedRadioButtonId() == -1){
                validationIsFine = false;
            }
        }
        if(field.getWidgetType() == SupportedWidgets.CALENDAR){
            LinearLayout dateLayout = (LinearLayout) field.getFieldView();
            EditText dateText = (EditText) dateLayout.getChildAt(0);
            if (dateText.getText() == null || dateText.getText().toString().isEmpty()) {
                validationIsFine = false;
            }
        }
        if(!validationIsFine){
            errorMsgs.append(Localization.translate("validation.required", field.getFieldView().getContext()));
        }
        return validationIsFine;
    }
}

File: AFMobile/AFWinPhone/AFWinPhone/components/parts/validators/RequiredValidator.cs
using AFWinPhone.enums;
using AFWinPhone.utils;
using System;
using System.Text;
using Windows.UI.Xaml.Controls;

namespace AFWinPhone.components.parts.validators
{
    class RequiredValidator : AFValidator
    {
        public bool validate(AFField field, StringBuilder errorMsgs, ValidationRule rule)
        {
            bool validationIsFine = true;
            if (Utils.IsFieldWritable(field.getFieldInfo().getWidgetType()))
            {
                TextBox textfield = (TextBox) field.getFieldView();
                if (String.IsNullOrWhiteSpace(textfield.Text))
                {
                    validationIsFine = false;
                }
            }
            if (field.getFieldInfo().getWidgetType().Equals(SupportedWidgets.PASSWORD))
            {
                PasswordBox passwordField = (PasswordBox)field.getFieldView();
                if (String.IsNullOrWhiteSpace(passwordField.Password))
                {
                    validationIsFine = false;
                }
            }
            if (field.getFieldInfo().getWidgetType().Equals(SupportedWidgets.OPTION))
            {
                StackPanel radioGroup = (StackPanel)field.getFieldView();
                bool atLeastOneIsChecked = false;
                foreach(RadioButton child in radioGroup.Children)
                {
                    if(child.IsChecked.Value)
                    {
                        atLeastOneIsChecked = true;
                    }
                }
                if (!atLeastOneIsChecked)
                {
                    validationIsFine = false;
                }
            }
            if (field.getFieldInfo().getWidgetType().Equals(SupportedWidgets.DROPDOWNMENU))
            {
                ComboBox cb = (ComboBox)field.getFieldView();
                if (cb.SelectedItem == null)
                {
                    validationIsFine = false;
                }
            }
            if (field.getFieldInfo().getWidgetType().Equals(SupportedWidgets.CALENDAR))
            {
                DatePicker textfield = (DatePicker)field.getFieldView();
                if (String.IsNullOrWhiteSpace(textfield.Date.ToString()))
                {
                    validationIsFine = false;
                }
            }
            if (!validationIsFine)
            {
                errorMsgs.Append(Localization.translate("validation.required"));
            }
            return validationIsFine;
        }
    }
}
